plugins/workspace-backpack/src/backpack.js

```diff
@@ -440,14 +440,39 @@ export class Backpack extends Blockly.DragTarget {
       this.addBlock(/** @type {!Blockly.BlockSvg} */ (dragElement));
     }
   }
+
   /**
-   * Converts the provided block into a cleaned XML string.
+   * Converts the provided block into a JSON string and
+   * cleans the JSON of any unnecessary attributes
    * @param {!Blockly.Block} block The block to convert.
-   * @returns {string} The cleaned XML string.
+   * @returns {string} The JSON object as a string
    * @private
    */
-  blockToCleanXmlString_(block) {
-    return cleanBlockXML(Blockly.Xml.blockToDom(block));
+  blockToJsonString(block) {
+    const json = Blockly.serialization.blocks.save(block);
+
+    // The keys to remove
+    const keys = ['id', 'height', 'width', 'pinned', 'enabled'];
+
+    // Traverse the JSON recursively
+    const traverseJson = function(json, keys) {
```

Comment:
Love the local function definition =) The fun of working in modern languages! (if JS still counts :P)